src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ITextSnapshotExtensions.cs

```diff
@@ -20,5 +22,33 @@ public static void GetLineAndCharacter(this ITextSnapshot snapshot, int index, o
             lineNumber = line.LineNumber;
             characterIndex = index - line.Start.Position;
         }
+
+        public static TextExtent? GetWordExtent(
+            this ITextSnapshot snapshot,
```

Comment:
null check the snapshot at the top of this method, ditto with the `textStructureNavigatorService`